### PR TITLE
Align KTO with DPO: Align _prepare_dataset

### PR DESCRIPTION
**Pull request by albertvillanova**
Align KTO with DPO: Align `_prepare_dataset`.

This PR refactors the dataset preparation logic in the `KTOTrainer` class to improve code maintainability and reduce duplication. The main changes involve extracting repeated dataset processing steps into a new helper method, centralizing KL dataset handling, and enforcing batch size requirements for KL calculation.

Part of:
- #4786

### Changes

**Refactoring and code organization:**

* Extracted all dataset preparation logic (prompt extraction, unpairing, chat template application, tokenization, KL dataset creation, and desirability balance checks) into a new `_prepare_dataset` method in `trl/experimental/kto/kto_trainer.py`, reducing code duplication and improving maintainability.
* Updated the main training and evaluation dataset setup to use the new `_prepare_dataset` method, simplifying the main training loop and ensuring consistent dataset processing for both train and eval datasets.

**Validation and error handling:**

* Added an explicit check to raise a `ValueError` if `per_device_train_batch_size` is less than or equal to 1 when KL calculation is enabled, ensuring correct KL term computation and preventing silent errors during training.

---

> [!NOTE]
> **Medium Risk**
> Touches KTO training data preprocessing and KL dataset construction, which can change training/eval inputs and fail fast for previously accepted configs (batch size <= 1). The changes are mostly refactor, but impact a core training path.
> 
> **Overview**
> Moves duplicated dataset preprocessing in `KTOTrainer` into a new `_prepare_dataset()` helper, and updates init to run both train and eval datasets through this single path (prompt extraction, unpairing, chat templating, tokenization, optional KL dataset creation/merge).
> 
> Adds an early `ValueError` when KL is enabled but `per_device_train_batch_size <= 1`, preventing miscomputed KL/implied-reward behavior; desirability balance warnings are now scoped to the training dataset only.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e10c4f202544d91075d0df0e054d7ffdc761e9d8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/kto/kto_trainer.py [hunk 1]
```diff
@@ -753,6 +617,110 @@ def make_inputs_require_grad(module, input, output):
...
                     )
 
+    def _prepare_dataset(
+        self,
+        dataset: Dataset,
+        processing_class: PreTrainedTokenizerBase | BaseImageProcessor | FeatureExtractionMixin | ProcessorMixin,
+        args: KTOConfig | None,
+        dataset_name: str,
+    ) -> Dataset:
+        tokenizer = getattr(processing_class, "tokenizer", processing_class)
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply trl/experimental/kto/kto_trainer.py:629]: This should be fixed once `_tokenizer` attribute is set.

**Review by qgallouedec — APPROVED** [review]
Lgtm
